apps/src/eval/eval.js

```diff
@@ -318,6 +318,30 @@ Eval.haveCaseMismatch_ = function (object1, object2) {
 };
 
 /**
+ * Note: is unable to distinguish from true/false generated from string blocks
+ *   vs. from boolean blocks
+ * @returns True if two eval objects are both booleans, but have different values.
+ */
+Eval.haveWrongBoolean_ = function (object1, object2) {
```

Comment:
Super subtle nit, I tend to try to create positive versions of helper methods like this to reduce the likelihood of double-negatives in future usages, which can become hard to parse if the logic ends up becoming more complex (e.g., some future flipped usage might be `else if (!haveWrongBoolean_("false", "true"))` versus a positive version `else if (booleansMatch("false", "true"))`

Maybe one simple way for this case without having to flip around usages and tests could be renaming to e.g. `haveBooleanMismatch_`.